native/src/main/java/io/ballerina/stdlib/http/api/service/signature/builder/AbstractPayloadBuilder.java

```diff
@@ -61,15 +64,19 @@ public static AbstractPayloadBuilder getBuilder(String contentType, Type payload
             return getBuilderFromType(payloadType);
         }
         contentType = contentType.toLowerCase(Locale.getDefault()).trim();
-        if (contentType.matches(XML_PATTERN)) {
+        BObject mediaType = MimeUtil.parseMediaType(ValueCreatorUtils.createMediaTypeObject(), contentType);
```

Comment:
Looks like we are creating unnecessary entity object. Shall we use the `HeaderUtil.getHeaderValue() method to get the base type of this content type header?